src/app/shared/wysiwyg/wysiwyg.component.ts

```diff
@@ -0,0 +1,392 @@
+/// <reference path="./prosemirror.d.ts"/>
+
+import { Component, Input, OnInit, OnChanges, OnDestroy, SimpleChanges, ElementRef, ViewChild } from '@angular/core';
+import { wrapIn, setBlockType, chainCommands, newlineInCode, toggleMark, baseKeymap } from 'prosemirror-commands';
+import { blockQuoteRule, orderedListRule, bulletListRule, codeBlockRule, headingRule,
+  inputRules,allInputRules } from 'prosemirror-inputrules';
+import { keymap } from 'prosemirror-keymap';
+import { schema, defaultMarkdownParser, defaultMarkdownSerializer } from 'prosemirror-markdown';
+import { MenuBarEditorView, toggleMarkItem, icons,
+  MenuItem, Dropdown, DropdownSubmenu,
+  wrapItem, blockTypeItem, joinUpItem, liftItem,
+  selectParentNodeItem} from 'prosemirror-menu';
+import { wrapInList, splitListItem, liftListItem, sinkListItem } from 'prosemirror-schema-list';
+import { EditorState, Plugin } from 'prosemirror-state';
+import { BaseModel } from '../model/base.model';
+import { ImageModel } from '../model/image.model';
+import { PromptComponent } from './prompt.component';
+
+@Component({
+  selector: 'publish-wysiwyg',
+  template: `
+    <div #contentEditable></div>
+    <publish-prompt #prompt>
+      <h1 class="modal-header">Link to</h1>
+      <div class="modal-body">
+        <label>URL<span class="error" [ngStyle]="{display: isURLInvalid() ? 'inline' : 'none'}">*</span></label>
```

Comment:
It's getting ng-invalid when empty because of the required attribute. We could use something like validatejs to check the URL https://validatejs.org/#validators-url